spec/unit/device_spec.lua

```diff
@@ -195,44 +185,44 @@ describe("device module", function()
         end)
     end)
 
-    describe("kindle #notest #nocov", function()
+    describe("kindle", function()
         it("should initialize voyage without error", function()
-            package.loaded['ffi/framebuffer_mxcfb'] = mock_fb
-            stub(io, "open")
-            io.open.returns({
-                read = function()
-                    return "XX13XX"
-                end,
-                close = function() end
-            })
-            mock_input = require('device/input')
-            stub(mock_input, "open")
+            io.open = function(filename, mode)
```

Comment:
We cannot stub `io.open` directly, it seems that `require` function use `io.open` to read module content so that subsequent module loading are all affected.